### PR TITLE
Change featureflags package to make it test friendly

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes:  https://github.com/open-telemetry/opentelemetry-collector/issues/4917

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +26,7 @@
 
 ### 🚩 Deprecations 🚩
 
+<<<<<<< HEAD
```
Comment (dmitryax) [inline CHANGELOG.md:29]: Merge conflict

**Review by dmitryax — COMMENTED** [review]
The approach LGTM , just one nit

**Review by codeboten — APPROVED** [review]
The approach makes sense to me 👍 

@dmitryax: the changelog has been fixed, PTAL

File: CHANGELOG.md (hunk 1 above)
Comment (bogdandrutu) [reply CHANGELOG.md]: fixed by @codeboten

**Review by dmitryax — APPROVED** [review]

**Pull request merged**